include/deal.II/base/table.h

```diff
@@ -2367,6 +2624,168 @@ Table<2,T>::n_cols () const
 
 
 
+//---------------------------------------------------------------------------
+namespace TransposeTableIterators
+{
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase()
+    :
+    container(nullptr),
+    linear_index(std::numeric_limits<decltype(linear_index)>::max())
+  {}
+
+
+
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase(const container_pointer_type table)
+    :
+    container(table),
+    linear_index(container->values.size())
+  {}
+
+
+
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase(const AccessorBase<T, false> &a)
+    :
+    container(a.container),
+    linear_index(a.linear_index)
+  {}
+
+
+
+  template <typename T, bool Constness>
+  inline
+  AccessorBase<T, Constness>::AccessorBase(const container_pointer_type table,
+                                           const std::ptrdiff_t         index)
+    :
+    container(table),
+    linear_index(index)
+  {}
+
+
+
+  template <typename T, bool Constness>
+  inline
+  const T &
+  AccessorBase<T, Constness>::value() const
+  {
+    Assert(0 <= this->linear_index,
+           ExcMessage("The current iterator points outside of the table."));
+    return this->container->values[this->linear_index];
```

Comment:
Don't you here know the type of the container and could ensure that the linear index is valid?

Reply:
We can add the check. I skipped it since it is technically redundant (`AlignedVector` will check) but I will add it back in.

Reply:
Oh, I had not realized that. I thought that `values` is just some `std::vector` or `double*` or something.

But it probably doesn't hurt either to check here already, rather than further down.